pytensor/tensor/subtensor.py

```diff
@@ -2686,6 +2687,32 @@ def grad(self, inputs, grads):
 advanced_subtensor = AdvancedSubtensor()
 
 
+@_vectorize_node.register(AdvancedSubtensor)
+def vectorize_advanced_subtensor(op: AdvancedSubtensor, node, *batch_inputs):
+    x, *idxs = node.inputs
+    batch_x, *batch_idxs = batch_inputs
+
+    # If indexes are batched fallback to Blockwise
+    if any(
+        batch_idx.type.ndim > idx.type.ndim
+        for batch_idx, idx in zip(batch_idxs, idxs)
+        if isinstance(batch_idx, TensorVariable)
+    ):
+        # Blockwise doesn't accept None or Slices types so we raise informative error here
+        # TODO: Implement these internally, so Blockwise is always a safe fallback
+        if any(not isinstance(idx.type, TensorVariable) for idx in idxs):
+            raise NotImplementedError(
+                "Vectorized AdvancedSubtensor combining batched indexes and slices or newaxis not supported."
+            )
+        else:
+            return vectorize_node_fallback(op, node, batch_x, *batch_idxs)
```

Comment:
This condition wasn't covered by a test. I think you should try to add a test that leads here.

Reply:
How dare you use codecov against me!!! (will do)

Reply:
All of this token nonsense had to bear some fruits!